### PR TITLE
Fixes ads enabled by default on upgrade of 1.2 to 1.3 for new ad regions

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave/brave-browser/issues/7741

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

Recommend tests:

Following upgrade paths from 0.63, 0.67, 0.68, 0.69, 0.70, 0.71, 0.72, 1.0, 1.1 and 1.2

- Test upgrade path for unsupported locale (on above versions) to supported locale on 1.3.x
- Test upgrade path for unsupported locale (on above versions) to unsupported locale on 1.3.x
- Test upgrade path for supported locale (on above versions, with Brave Ads enabled) to supported locale on 1.3.x

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/extensions/api/brave_rewards_api.cc [hunk 1]
```diff
@@ -605,7 +605,9 @@ ExtensionFunction::ResponseAction BraveRewardsSaveAdsSettingFunction::Run() {
   AdsService* ads_service_ = AdsServiceFactory::GetForProfile(profile);
   if (ads_service_) {
     if (params->key == "adsEnabled") {
-      ads_service_->SetEnabled(params->value == "true");
+      const auto is_enabled =
+          params->value == "true" && ads_service_->IsSupportedLocale();
```
Comment (tmancey) [inline browser/extensions/api/brave_rewards_api.cc]: The bug was caused by this line, due to a legacy issue where the ads flag was enabled even if the users locale was unsupported

**Comment by tmancey** [comment]
@NejcZdovc @gdregalo Unrelated CI failure

**Comment by bsclifton** [comment]
Helping merge since code owner unavailable. CI looks good- it's yellow (acceptable - ran into problem on network audit + test install. there are known issues). Good to merge if otherwise approved 👍

**Comment by LaurenWags** [comment]
Verified using

Brave | 1.5.57 Chromium: 79.0.3945.130 (Official Build) nightly (64-bit)
-- | --
Revision | e22de67c28798d98833a7137c0e22876237fc40a-refs/branch-heads/3945@{#1047}
OS | macOS Version 10.14.6 (Build 18G103)

* Verified test plan from description:

1. Upgrade path for unsupported locale to supported locale
   - **FAIL** When upgrading from 0.63.55 or 0.64.77 I encounter https://github.com/brave/brave-browser/issues/7833#issuecomment-577916890
   - **PASS** Installed 0.65.121 for Spain. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.
   - **PASS** Installed 0.66.101 for Spain. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region. Enabled Ads and confirmed I received an ad.
   - **PASS** Installed 0.67.125 for Norway. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.
   - **PASS** Installed 0.68.142 for Norway. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.
   - **PASS** Installed 0.69.135 for Norway. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.
   - **PASS** Installed 0.70.123 for Norway. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.
   - **PASS** Installed 1.0.1 for Russia. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.
   - **PASS** Installed 1.1.23 for Russia. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.
   - Installed 1.2.43 for Russia. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: ads were not enabled by default and there was a BAT logo notification notifying the user that Brave Ads are now available in their region.

2. Upgrade path for unsupported locale to unsupported locale
   - **FAIL** When upgrading from 0.63.55 or 0.64.77 for Aruba I encounter https://github.com/brave/brave-browser/issues/7833#issuecomment-577916890
   - **PASS** Installed 0.65.121 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 0.66.101 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 0.67.125 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 0.68.142 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 0.69.135 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 0.70.123 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 1.0.1 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 1.1.23 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.
   - **PASS** Installed 1.2.43 for Aruba. Enabled Rewards. Confirmed “Sorry” message on Ads panel. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: still see "Sorry" message and there is no BAT logo encouraging user to try Ads.

3. Upgrade path for supported locale (with Ads enabled) to supported locale
   - **FAIL** When upgrading from 0.63.55 or 0.64.77 for US I encounter https://github.com/brave/brave-browser/issues/7833#issuecomment-577916890
   - **PASS** Installed 0.65.121 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 0.66.101 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 0.67.125 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 0.68.142 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 0.69.135 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 0.70.123 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 1.0.1 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 1.1.23 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.
   - **PASS** Installed 1.2.43 for US. Enabled Rewards and viewed an Ad. Upgraded to 1.5.x (renamed profile). After upgrade worked as expected: Ads still on after upgrade, Ads panel displays expected data, can still receive ad notifications.

**Pull request merged**